hazelcast/src/main/java/com/hazelcast/logging/Logger.java

```diff
@@ -71,16 +71,40 @@ public static ILogger getLogger(Class clazz) {
      * @return the obtained logger.
      */
     public static ILogger getLogger(String name) {
-        // try the fast path first
         LoggerFactory existingFactory = loggerFactory;
-        if (existingFactory != null) {
+        if (loggerFactory != null) {
             return existingFactory.getLogger(name);
         }
+        return createFactoryInternal(name).getLogger(name);
+    }
 
+    /**
+     * Removes the {@link ILogger logger} for the given {@code clazz}.
+     *
+     * @param clazz the class of the logger to remove.
+     */
+    public static void removeLogger(Class clazz) {
+        removeLogger(clazz.getName());
```

Comment:
nit: it would be nice to have input validation (e.g. `checkNotNull(clazz, "class must not be null");`) in this and other public methods of this class.